domain/src/test/java/org/oppia/domain/UserAppHistoryControllerTest.kt

```diff
@@ -115,25 +116,24 @@ class UserAppHistoryControllerTest {
 
   @Test
   @ExperimentalCoroutinesApi
-  fun testControllerObserver_observedAfterSettingAppOpened_providesLiveData_userDidNotOpenApp()
-      = runBlockingTest(coroutineContext) {
-    val appHistory = userAppHistoryController.getUserAppHistory()
-
-    appHistory.observeForever(mockAppHistoryObserver)
-    userAppHistoryController.markUserOpenedApp()
-    advanceUntilIdle()
-
-    // The result should not indicate that the user opened the app because markUserOpenedApp does not notify observers
-    // of the change.
-    verify(mockAppHistoryObserver, atLeastOnce()).onChanged(appHistoryResultCaptor.capture())
-    assertThat(appHistoryResultCaptor.value.isSuccess()).isTrue()
-    assertThat(appHistoryResultCaptor.value.getOrThrow().alreadyOpenedApp).isFalse()
-  }
+  fun testControllerObserver_observedAfterSettingAppOpened_providesLiveData_userDidNotOpenApp() =
```

Comment:
Ditto test is not working.

Reply:
test is not working means it's not passing  ?

Reply:
It throws few errors related to observer.

Reply:
Reverted this changes but still you will face the same issue. Will discuss about the same with ben.